src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs

```diff
@@ -152,7 +153,7 @@ private async Task BindOutputsAsync(object input, Binder binder, ScriptInvocatio
 
             _handleScriptReturnValue(result);
 
-            var outputBindingTasks = _outputBindings.Select(async binding =>
+            var outputBindingTasks = _outputBindings.AsParallel().Select(async binding =>
```

Comment:
I do not think we have E2E tests as part of PR builds that large number of multiple inputs/outputs. Can you please update the PR description with the tests you have run and any perf improvements this changes would bring in?

Reply:
cc @brettsam / @mathewc  - if they have any concerns processing multiple output bindings in parallel

Reply:
I added the description of the test application I used for validating this change.
My tests show that the behavior we get with this change is pretty much the optimal one would expect.
The more bindings we have, the more functions benefit from this.
However, as mentioned in the other comment we may want to limit the maximum number of parallel requests.

Reply:
We already bind the input/output bindings in parallel - that's what the below Task.WhenAll calls do. So I don't see why we need any changes here?


This is for Functions out of proc. Interestingly, the core SDK has always bound inputs/outputs serially (input binding code [here](https://github.com/Azure/azure-webjobs-sdk/blob/6e2e5695ced990ae32b01d8108777bd87c7b7f22/src/Microsoft.Azure.WebJobs.Host/Triggers/TriggeredFunctionBinding.cs#L83), output binding code [here](https://github.com/Azure/azure-webjobs-sdk/blob/b05b4466ed67de8331dd940e1262b5efb43c9f65/src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs#L933)).

Reply:
@mathewc from the code, that's what I thought too.
However, when running this and timing it, I see the bindings been done sequentially.
It must be some weird behavior with the await/async.

Reply:
Should be straight forward to write a unit test that verifies that these bindings are executed in parallel. The Linq query returns an enumerable of tasks that may have started sequentially, but they'll complete at different times based on what they do I expect. I'd like to see a test proving that this isn't happening, since it's hard to believe :) We use Task.WhenAll all over the codebase for parallel execution.

Reply:
I was as skeptical as you :)
We tested adding {{await Task.Delay(1000)}} but in this case, it does run in parallel.
The issue comes when the code inside runs without stop (e.g., the binding one).
I am not very familiar with the unit testing infrastructure for the bindings, do you have any suggestion?
I can also do a screen share session with the example app and the timing.

Reply:
This is happening with the Azure Blob binding.
For blocking I was referring anything that does not do Delay().
I can add instrumentation to the code and show the timings before and after.